ST417 Bayesian Modelling Project.Rmd

```diff
@@ -67,49 +71,217 @@ scale_and_avg_range <- function(column, to_match, scale) {
+
+# prior and posterior comparison
+prior_post_plot(bayes_df_time)
+```
+
+```{r distance prior}
+theta_dist <- na.omit(data$est_distance)
+prior_dist <- dnorm(theta_dist, mean = mean(theta_dist), sd = sd(theta_dist))
+bayes_df_dist <- data.frame(theta_dist, prior_dist)
+bayes_df_dist
+prob_plot(bayes_df_dist)
+```
+
+```{r distance posterior}
+xbar_dist <- mean(data$distance, na.rm = TRUE)
+n_dist <- NROW(data$distance)
+sigma_dist <- sd(data$distance, na.rm = TRUE)
+se_dist <- sigma_dist / sqrt(n_dist)
+
+# likelihood calculation
+likelihood_dist <- dnorm(xbar_dist, mean = theta_dist, sd = se_dist)
+bayes_df_dist <- data.frame(theta_dist, prior_dist, likelihood_dist)
+
+# posterior calculation
+bayes_df_dist$Product <- bayes_df_dist$prior_dist * bayes_df_dist$likelihood_dist # nolint
+bayes_df_dist$Posterior <- bayes_df_dist$Product / sum(bayes_df_dist$Product)
+bayes_df_dist
+
+# prior and posterior comparison
+prior_post_plot(bayes_df_dist)
+```
+
+```{r cost prior}
+theta_cost <- na.omit(data$est_cost)
+prior_cost <- dnorm(theta_cost, mean = mean(theta_cost), sd = sd(theta_cost))
+bayes_df_cost <- data.frame(theta_cost, prior_cost)
+bayes_df_cost
+prob_plot(bayes_df_cost)
+```
+
+```{r cost posterior}
+xbar_cost <- mean(data$cost, na.rm = TRUE)
+n_cost <- NROW(data$cost)
+sigma_cost <- sd(data$cost, na.rm = TRUE)
+se_cost <- sigma_cost / sqrt(n_cost)
+
+# likelihood calculation
+likelihood_cost <- dnorm(xbar_cost, mean = theta_cost, sd = se_cost)
+bayes_df_cost <- data.frame(theta_cost, prior_cost, likelihood_cost)
+
+# posterior calculation
+bayes_df_cost$Product <- bayes_df_cost$prior_cost * bayes_df_cost$likelihood_cost # nolint
+bayes_df_cost$Posterior <- bayes_df_cost$Product / sum(bayes_df_cost$Product)
+bayes_df_cost
+
+# prior and posterior comparison
+prior_post_plot(bayes_df_cost)
+```
+
+```{r}
+m0_time <- mean(data$est_time, na.rm = TRUE)
+m0_dist <- mean(data$est_distance, na.rm = TRUE)
+m0_cost <- mean(data$est_cost, na.rm = TRUE)
+s0_time <- sd(data$est_time, na.rm = TRUE)
+s0_dist <- sd(data$est_distance, na.rm = TRUE)
+s0_cost <- sd(data$est_cost, na.rm = TRUE)
+```
+
+```{r}
+post_mean_time <- ((1 / (se_time^2) + 1 / (s0_time^2))^(-1)) * (xbar_time / (se_time^2) + m0_time / (s0_time^2)) # nolint
+post_sd_time <- sqrt((1 / (se_time^2) + 1 / (s0_time^2))^(-1))
+
+post_mean_dist <- (1 / (se_dist^2) + 1 / (s0_dist^2))^(-1) * (xbar_dist / (se_dist^2) + m0_dist / (s0_dist^2)) # nolint
+post_sd_dist <- sqrt((1 / (se_dist^2) + 1 / (s0_dist^2))^(-1))
+
+post_mean_cost <- (1 / (se_cost^2) + 1 / (s0_cost^2))^(-1) * (xbar_cost / (se_cost^2) + m0_cost / (s0_cost^2)) # nolint
+post_sd_cost <- sqrt((1 / (se_cost^2) + 1 / (s0_cost^2))^(-1))
+```
+
+```{r 95 credible intervals}
+# 95% credible intervals
+time_sims <- rnorm(10000, mean = post_mean_time, sd = post_sd_time)
+time_point_estimate <- mean(time_sims)
+time_point_estimate
+
+time_ci95 <- quantile(time_sims, probs = c(0.025, 0.975))
+time_ci95
+normal_interval(0.95, time_ci95)
```

Comment:
The credible intervals fall into the negative range. I think this is because the sd is large, and the distribution is not bounded at zero. We will need to fix this in a future update.

Reply:
Could you also please label these plots? It's not clear whether this is the posterior for cost or average cost.